src/signer/canister/tests/it/utils/test_environment.rs

```diff
@@ -0,0 +1,199 @@
+    /// ICRC2 ledger
+    pub ledger: CyclesLedgerPic,
+    /// User
+    pub user: Principal,
+    /// Another user
+    pub user2: Principal,
+    /// A crowd
+    pub users: [Principal; 5],
+    /// Unauthorized user
+    pub unauthorized_user: Principal,
+    /// A canister used to deposit cycles into the ledger.
+    pub cycles_depositor: CyclesDepositorPic,
+}
+impl Default for TestSetup {
+    fn default() -> Self {
+        let pic = Arc::new(
+            PocketIcBuilder::new()
+                .with_fiduciary_subnet()
+                .with_system_subnet()
+                .with_application_subnet()
+                .with_ii_subnet()
+                .with_nns_subnet()
+                .build(),
+        );
+        let cycles_ledger_canister_id = pic
+            .create_canister_with_id(None, None, cycles_ledger_canister_id())
+            .unwrap();
+
+        // Would like to create this with the cycles ledger canister ID but currently this yields an error.
+        let ledger = CyclesLedgerPic::from(
+            PicCanisterBuilder::default()
+                .with_canister(cycles_ledger_canister_id)
+                .with_wasm(&dfx_wasm_path("cycles_ledger"))
+                .with_arg(
+                    encode_one(LedgerArgs::Init(LedgerInitArgs {
+                        index_id: None,
+                        max_blocks_per_request: 999,
+                    }))
+                    .expect("Failed to encode ledger init arg"),
+                )
+                .deploy_to(pic.clone()),
+        );
+        let signer = SignerPic::from(
+            PicCanisterBuilder::default()
+                .with_wasm(&cargo_wasm_path("signer"))
+                .with_arg(
+                    encode_one(Arg::Init(InitArg {
+                        ecdsa_key_name: format!("test_key_1"),
+                        ic_root_key_der: None,
+                        cycles_ledger: None,
+                    }))
+                    .unwrap(),
+                )
+                .deploy_to(pic.clone()),
+        );
+        let user = Principal::from_text(CALLER).unwrap();
+        let user2 =
+            Principal::from_text("jwhyn-xieqy-drmun-h7uci-jzycw-vnqhj-s62vl-4upsg-cmub3-vakaq-rqe")
+                .unwrap();
+        let users = [
+            Principal::from_text("s2xin-cwqnw-sjvht-gp553-an54g-2rhlc-z4c5d-xz5iq-irnbi-sadik-qae")
+                .unwrap(),
+            Principal::from_text("dmvof-2tilt-3xmvh-c7tbj-n3whk-k2i6b-2s2ge-xoo3d-wjuw3-ijpuw-eae")
+                .unwrap(),
+            Principal::from_text("kjerd-nj73t-u3hhp-jcj4d-g7w56-qlrvb-gguta-45yve-336zs-sunxa-zqe")
+                .unwrap(),
+            Principal::from_text("zxhav-yshtx-vhzs2-nvuu3-jrq66-bidn2-put3y-ulwcf-2gb2o-ykfco-sae")
+                .unwrap(),
+            Principal::from_text("nggqm-p5ozz-i5hfv-bejmq-2gtow-4dtqw-vjatn-4b4yw-s5mzs-i46su-6ae")
+                .unwrap(),
+        ];
+        let unauthorized_user =
```

Comment:
Unauthorized for what?

I see this is a controller of the cycles depositor. Is that related to being unauthozied?

Reply:
This is the user I use whenever I need an external bad guy in a test.  Like Eve the evil persona in a communication between Alice and Bob.  Random third parties should not be able to steal resources or break th eprotocol.

Reply:
Added a comment to that effect.